pandas/tests/frame/test_constructors.py

```diff
@@ -2245,6 +2246,31 @@ def test_from_records_empty_with_nonempty_fields_gh3682(self):
         tm.assert_index_equal(df.index, Index([], name="id"))
         assert df.index.name == "id"
 
+    @pytest.mark.parametrize(
+        "input_value, expected_value",
+        [
+            ("int16", np.dtype("int16")),
+            ("int32", np.dtype("int32")),
+            ("int64", np.dtype("int64")),
+            ("float16", np.dtype("float16")),
+            ("float32", np.dtype("float32")),
+            ("float64", np.dtype("float64")),
+            ("uint16", np.dtype("uint16")),
+            ("uint32", np.dtype("uint32")),
+            ("uint64", np.dtype("uint64")),
+            ("datetime64[ns]", np.dtype("<M8[ns]")),
+            ("timedelta64[ns]", np.dtype("<m8[ns]")),
+            ("bool", np.dtype("bool")),
+            ("object", np.dtype("O")),
+            ("category", CategoricalDtype(categories=[], ordered=False)),
```

Comment:
great as long as you are adding a list, can you add the rest of the types. you ca easily do this by using the types defined in pandas._testing (we also have fixtures for this, but maybe not one that covers *all* types); would be ok with one for that as well. Note that I don't think we have Boolean, String defined pandas._testing (could add them there)

Reply:
Added more dtypes check, but now the list is very long. Any ideas on how to makes more pretty? I did not want to add it to the lists in pandas._testing, since I was not sure it would be compatible with other tests that rely on them.

Reply:
you don't need to do it like this, rather use the *already* defined ones that are ther
e.g. ALL_EA_INT_DTYPES

Reply:
Do you have any tips on how to refactor the parametrization if I use ALL_EA_INT_DTYPES? The parametrization now has the form [(input_value, expected_value)], but if I loop over ALL_EA_INT_DTYPES I think I still need to manually provide the expected value every time.

Reply:
how's that? the expected IS the same as the input

Reply:
Not always. Below are some examples in the form "input: expected". The transformation is also not always the same (applying np.dtype does not work)
"uint8", np.dtype("uint8")
"UInt8", UInt8Dtype()
int, np.dtype("int64")
"Int8", Int8Dtype()

Reply:
of course, you can easily tranform these: 

```
In [1]: pd.Int64Dtype().name                                                                                                                                                                                        
Out[1]: 'Int64'
```

Reply:
so my points is that you can simply pass in the ALL_* (just concatenate them all), and check the dtypes of the result columns

Reply:
Great, that already helps me a lot. But I for `int` this would not work since the expected is 'Int64'. Similarly for `float`, `str`. 

Sorry for the large number of questions, this is my first PR :)

Reply:
sure so what you can do is break this into 2 tests, one for the easy cases where this is true and one for the special cases (where you can input 2 args for input and expected)

Reply:
Thanks for the feedback, I think I came to a much better test now.